src/combineReducers.js

```diff
@@ -113,11 +113,9 @@ export default function combineReducers(reducers) {
       throw sanityError
     }
 
-    if (process.env.NODE_ENV !== 'production') {
-      var warningMessage = getUnexpectedStateShapeWarningMessage(state, finalReducers, action)
-      if (warningMessage) {
-        warning(warningMessage)
-      }
+    var warningMessage = getUnexpectedStateShapeWarningMessage(state, finalReducers, action)
```

Comment:
Hmm. We definitely don’t want to execute this code in production path, it’s too slow.
Can you look into a different way of solving it?